### ISSUE TITLE
init() not firing when existing element gets matching class added

### ISSUE THREAD

**Issue by davidmerrique** [WebReflection/wicked-elements#34]
Hey, loving the changes to v1.

I found some behavior that changed.

## v0.9.1 behavior
```html
<div class="to-become-wicked">Not Wicked</div>
```
```javascript
wickedElements.define('.wicked', {
  init(event) {
    this.element = event.currentTarget;
    this.element.textContent = 'I am wicked';
  }
});

document.querySelector('.to-become-wicked').classList.add('wicked');
```
outputs:
```html
<div class="to-become-wicked wicked">I am wicked</div>
```

## v1.0.6 behavior
```javascript
wickedElements.define('.wicked', {
  init() {
    this.element.textContent = 'I am wicked';
  }
});

document.querySelector('.to-become-wicked').classList.add('wicked');
```
Doesn't do anything:
```html
<div class="to-become-wicked wicked">Not Wicked</div>
```

`init() {...}` doesn't fire!

Not sure if that was intentional.

Thank you for this great tool!

**Comment by WebReflection** [comment]
the definition searches for live nodes and doesn't find any, so the question is why would you need any indirection in between ... just use the class you know it'll be wicked, or use the `upgrade` mechanism.

```js
for (const wicked of document.querySelectorAll('.to-become-wicked')) {
  wicked.classList.add('wicked');
  wickedElements.upgrade(wicked);
}
```

Doing a search through the whole document in intervals, or listening to all possible classes changes per each node in any page, is a performance hazard, shenanigan, killer, but also an edge/non-common case.

However, you could also define `.to-become-wicked` as such:

```js
const {define, upgrade, whenDefined} = wickedElements;
define('.to-become-wicked', {
  init() {
    whenDefined('.wicked').then(upgrade.bind(null, this.element));
  }
});
```

Basically, you use a behavior to pass the element to another behavior once defined, so you have all primitives to do whatever you want, and I am not planning to fix this anyhow, for the already mentioned reasons regarding performances.

`upgrade` is also safe to use for already upgraded elements, as everything is handled by reference, so there won't ever be duplicated upgrades per behavior.

I hope this answers your question, happy to expand more if needed.

**Comment by davidmerrique** [comment]
Alright, I'll use `upgrade`.

~~I think the readme should explain `upgrade`.~~ I just noticed [`customElements.upgrade`](https://html.spec.whatwg.org/multipage/custom-elements.html#dom-customelementregistry-upgrade)

My use case is something like this, which just worked in previous versions (I only want enhanced behavior at a specific breakpoint):
```javascript
wickedElements.define('.enhanced', {...});

if (window.matchMedia('(min-width: 768px)').matches) {
  const basic = document.querySelector('.basic-element');
  basic.classList.add('enhanced');
}
```

So now in v1 I have to do this:
```javascript
wickedElements.define('.enhanced', {...});

if (window.matchMedia('(min-width: 768px)').matches) {
  const basic = document.querySelector('.basic-element');
  basic.classList.add('enhanced');
  wickedElements.upgrade(basic);
}
```

Thanks

**Comment by WebReflection** [comment]
That's one way to go, probably my pick too, but you could also define `.basic-element` and call it a day:

```js
if (matchMedia('(min-width: 768px)').matches)
  wickedElements.define('.basic-element', {
    init() {
      this.element.classList.add('enhanced');
      wickedElements.upgrade(this.element);
    }
  });
```

or move the condition within the `.basic-element` definition

**Comment by WebReflection** [comment]
Just FYI, I've added few **F.A.Q.** in the [README](https://github.com/WebReflection/wicked-elements#faq), covering the _My element doesn't become wicked, what should I do?_ case too, explaining how to upgrade all elements at once, after lazy changes/definitions, which can always be obtained via `wickedElements.upgrade(document.documentElement);` 👋

**Comment by davidmerrique** [comment]
@WebReflection Love it. Thank you.
